src/utils.ts

```diff
@@ -0,0 +1,46 @@
+import { Principal } from "@dfinity/principal";
+import { IndefiniteEvent, DetailValue } from "./declarations/cap/root";
+
+type DetailType =
+  | Principal
+  | bigint
+  | boolean
+  | string
+  | number
+  | Array<number>
+  | Array<DetailType>;
+type PrettyDetail = { [key: string]: DetailType };
+type DetailVector = { Vec: Array<DetailValue> };
+type GeneralValue = { [key: string]: any };
+
+const decodeDetailValue = (value: DetailValue): DetailType => {
+  const type = Object.keys(value)?.[0];
+  switch (type) {
+    case "Vec":
+      return (value as DetailVector)?.["Vec"]?.map(decodeDetailValue);
+    case "True":
+      return true;
+    case "False":
+      return false;
+    default:
+      return (value as GeneralValue)?.[type];
+  }
+};
+
+export const preattifyCapTransactions = (transaction: IndefiniteEvent) => {
+  const details = transaction?.details?.reduce<PrettyDetail>(
+    (acum, [key, value]) => ({
+      ...acum,
+      [key]: decodeDetailValue(value),
+    }),
+    {} as PrettyDetail
+  );
+  return {
+    details,
+    to: details?.to,
+    from: details?.from,
+    caller: transaction.caller,
+    operation: transaction.operation,
+    // time: transaction.time // TODO: Uncomment this when candid is updated
```

Comment:
This can be uncommented from now, but to be fair, can be used straight at the top-level, so maybe not need to be passed here...

```suggestion
    time: transaction.time // TODO: Uncomment this when candid is updated
```